numba_cuda/numba/cuda/tests/nrt/test_nrt_refct.py

```diff
@@ -0,0 +1,118 @@
+
+import gc
+import numpy as np
+import unittest
+from numba.tests.support import override_config
+from numba.cuda.runtime import rtsys
+from numba.cuda.tests.support import EnableNRTStatsMixin
+from numba.cuda.testing import CUDATestCase
+from numba.cuda.tests.nrt.mock_numpy import cuda_empty, cuda_empty_like
+
+from numba import cuda
+
+
+class TestNrtRefCt(EnableNRTStatsMixin, CUDATestCase):
+
+    def setUp(self):
+        # Clean up any NRT-backed objects hanging in a dead reference cycle
+        gc.collect()
+        super(TestNrtRefCt, self).setUp()
+
+    def tearDown(self):
+        super(TestNrtRefCt, self).tearDown()
```

Comment:
I think these tests were ported from the CPU target - it's not clear to me that the logic applies here as we don't have any connection between objects on the CPU and allocations / refcounts on the GPU. Am I missing something?

Reply:
I don't think the `gc.collect()` bits are relavant here, but calling the parent ctor (from `EnableNRTStatsMixin` parent) is. This is a new mixin we created for this PR that enables the memsys stats on test startup and disables on shutdown.

Reply:
Removed in https://github.com/NVIDIA/numba-cuda/pull/78/commits/97b65b27725e968535d22c02201960d6a9aa8ac8.